crates/interpreter/src/valid.rs

```diff
@@ -764,17 +804,50 @@ impl<'a, 'm> Expr<'a, 'm> {
         Ok(())
     }
 
-    fn br_label(&self, l: LabelIdx) -> Result<ResultType<'m>, Error> {
+    fn stitch(&mut self, branch: &SideTableBranch) {
+        let current_branch = self.branch();
+        let SideTableBranch { parser, side_table } = branch;
+        let entry = &mut self.side_table[*side_table];
+        assert!(entry.is_none());
+        *entry = Some(SideTableEntryView {
+            delta_ip: SideTableBranch::delta_ip(current_branch.parser, parser),
+            delta_stp: SideTableBranch::delta_stp(current_branch.side_table, *side_table),
+            // TODO(dev/fast-interp): Compute the fields below.
+            val_cnt: 0,
+            pop_cnt: 0,
+        });
+    }
+
+    fn br_label(&mut self, l: LabelIdx) -> Result<ResultType<'m>, Error> {
         let l = l as usize;
         let n = self.labels.len();
         check(l < n)?;
-        let label = &self.labels[n - l - 1];
+        let branch = self.branch();
+        let label = &mut self.labels[n - l - 1];
         Ok(match label.kind {
-            LabelKind::Block | LabelKind::If => label.type_.results,
-            LabelKind::Loop => label.type_.params,
+            LabelKind::Block => {
+                label.branches.push(branch);
+                self.side_table.push(None);
+                label.type_.results
+            }
+            LabelKind::If => label.type_.results,
+            LabelKind::Loop(SideTableBranch { parser, side_table }) => {
+                self.side_table.push(Some(SideTableEntryView {
+                    delta_ip: SideTableBranch::delta_ip(branch.parser, parser),
+                    delta_stp: SideTableBranch::delta_stp(branch.side_table, side_table),
+                    // TODO(dev/fast-interp): Compute the fields below.
+                    val_cnt: 0,
+                    pop_cnt: 0,
+                }));
```

Comment:
It's probably cleaner to do something like this:
```suggestion
            LabelKind::Loop(target) => {
                self.stitch(source, target);
```

We need to make `stitch` take both source and target instead of using `self.branch()`. This is accidentally better, because then we can factorize the call to `self.branch()` in `fn end()`.